docs/Joins.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Joining Streams in Storm Core
+layout: documentation
+documentation: true
+---
+
+Storm core supports joining multiple data streams into one with the help of `JoinBolt`.
+`JoinBolt` is a Windowed bolt, i.e. it waits for the configured window duration to match up the
+tuples among the streams being joined. This helps align the streams within the Window boundary.
+
+Each of `JoinBolt`'s incoming data streams must be Fields Grouped on a single field. A stream 
+should only be joined with the other streams using the field on which it has been FieldsGrouped.  
+Knowing this will help understand the join syntax described below.  
+
+## Performing Joins
+Consider the following SQL join involving 4 tables called: stream1, stream2, stream3 & stream4:
+
+```sql
+select  userId, key4, key2, key3
+from    stream1 
+join       stream2  on stream2.userId =  stream1.key1
+join       stream3  on stream3.key3   =  stream2.userId
+left join  stream4  on stream4.key4   =  stream3.key3
+```
+
+This could be expressed using `JoinBolt` over 4 similarly named streams as: 
+
+```java
+new JoinBolt(JoinBolt.Selector.STREAM, "stream1", "key1")               // from stream1  
+                            .join     ("stream2", "userId",  "stream1") // join      stream2  on stream2.userId = stream1.key1
```

Comment:
the sql like syntax is not supported correct? So assume joining on subset of join fields (the "not ok" case in your example) is not possible with the current api. 
Its only about allowing users to have multiple fields as the join key without the extra merge step. May be accepting a type like `JoinKey` than string as the key parameter in the api would make it clear.